tests/utils.py

```diff
@@ -55,10 +61,26 @@ def venv_python(venv_dir: Path) -> Path:
     return venv_dir / "bin" / "python"
 
 
+# ====================================================================
+# Parsing the requirements file
+# ====================================================================
+
+
+REQS_FILE = "requirements-tests.txt"
+
+
 @cache
+def parse_requirements() -> dict[str, Requirement]:
```

Comment:
nit: returning a mutable type from a cached function is a little unsafe; I'd use `Mapping` here (needs an import from `collections.abc`):

```suggestion
def parse_requirements() -> Mapping[str, Requirement]:
```